external/fv3fit/fv3fit/testing.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, Hashable, Iterable, Mapping, Optional, Union
+from ._shared import Predictor, io
+import numpy as np
+import xarray as xr
+import os
+import yaml
+
+
+@io.register("constant-output")
+class ConstantOutputPredictor(Predictor):
+    """
+    A simple predictor meant to be used for testing.
+    
+    Supports scalar and vector outputs, where the vector outputs are all
+    of the same shape and assigned a dimension name of "z".
+    """
+
+    def __init__(
+        self,
+        sample_dim_name: str,
+        input_variables: Iterable[Hashable],
+        output_variables: Iterable[Hashable],
+    ):
+        """Initialize the predictor
+        
+        Args:
+            sample_dim_name: name of sample dimension
+            input_variables: names of input variables
+            output_variables: names of output variables
+        
+        """
+        super().__init__(
+            sample_dim_name=sample_dim_name,
+            input_variables=input_variables,
+            output_variables=output_variables,
+        )
+        self._outputs: Dict[Hashable, Union[np.ndarray, float]] = {}
+
+    def set_outputs(self, **outputs: Union[np.ndarray, float]):
+        """
```

Comment:
Slightly more info here would be helpful in writing a test using this., e.g., that the output variables will have dimensions (sample_dim_name, "z") if a 1-D array is passed and (sample_dim_name) if a scalar is passed.

Reply:
Added more description.